File: lib/ghez/ghez/time.ex
defmodule Ghez.Time do
  def convert_hours_duration_to_human(duration) do
    case convert_hours_to_hours_minutes(duration) do
      {0, minutes} ->
        "#{minutes} minutes"

      {hours, 0} ->
        "#{hours} hours"

      {hours, minutes} ->
        "#{hours} hours and #{minutes} minutes"
    end
  end

  def convert_hours_to_hours_minutes(duration) do
    hours = trunc(duration)
    minutes = ((duration - hours) * 60) |> trunc()
    {hours, minutes}
  end
end

File: assets/js/time.js
export function convertHoursDurationToHuman(duration) {
  const {
    hours = 0,
    minutes = 0,
  } = convertHoursDurationToHoursAndMinutesDuration(duration);

  const hoursText = `${hours} hours`;
  const minutesText = `${minutes} minutes`;

  if (hours && minutes) return `${hoursText} and ${minutesText}`;
  if (hours) return hoursText;
  return minutesText;
}

export function convertHoursDurationToHoursAndMinutesDuration(duration) {
  const hours = Math.floor(duration);
  const minutes = Math.floor((duration - hours) * 60);
  return {
    hours,
    minutes,
  };
}
